test/system/src/main/java/io/pravega/test/system/framework/kubernetes/K8sClient.java

```diff
@@ -547,20 +551,25 @@ public void deletePVC(String namespace, String name) {
      */
     public CompletableFuture<Void> downloadLogs(final V1Pod fromPod, final String toFile) {
 
-        return Retry.withExpBackoff(500, 10, LOG_DOWNLOAD_RETRY_COUNT, RETRY_MAX_DELAY_MS)
+        return Retry.withExpBackoff(LOG_DOWNLOAD_INIT_DELAY_MS, 10, LOG_DOWNLOAD_RETRY_COUNT, RETRY_MAX_DELAY_MS)
                     .retryingOn(TestFrameworkException.class)
                     .throwingOn(RuntimeException.class)
                     .runInExecutor(() -> {
                         final String podName = fromPod.getMetadata().getName();
-                        log.debug("copy logs from pod {} to file {}", podName, toFile);
+                        log.debug("Download logs from pod {}", podName);
                         try {
                             @Cleanup
                             InputStream logStream = logUtility.streamNamespacedPodLog(fromPod);
-                            Files.copy(logStream, Paths.get(toFile));
-                            log.debug("log copy completed for pod {}", podName);
+                            // On every retry this method attempts to download the complete pod logs from from K8s api-server. Due to the
+                            // amount of logs for a pod and the K8s cluster configuration it can so happen that the K8s api-server can
+                            // return truncated logs. Hence, every retry attempt does not overwrite the previously downloaded logs for
+                            // the pod.
+                            String logFile = toFile + "-" + randomAlphanumeric(5) + ".log";
```

Comment:
Instead of an alphanumeric, could we keep the order of logs (i.e., use a number sequence)? This would help people debugging to keep the sense of time while looking across logs.

Reply:
makes sense, done.